src/main/java/org/junit/rules/TemporaryFolder.java

```diff
@@ -205,9 +205,16 @@ public File newFolder(String... paths) throws IOException {
             file = new File(root, relativePath.getPath());
 
             lastMkdirsCallSuccessful = file.mkdirs();
-            if (!lastMkdirsCallSuccessful && !file.isDirectory()) {
-                throw new IOException(
-                        "could not create a folder with the path \'" + relativePath.getPath() + "\'");
+            if (!lastMkdirsCallSuccessful) {
+                if (file.exists() && file.isDirectory()) {
+                    continue;
```

Comment:
I think we could avoid the `continue` by moving the `!file.isDirectory()` back into the enclosing `if` and remove the first inner `if`.

Reply:
You're right. Fixed.